torch/utils/_cxx_pytree.py

```diff
@@ -193,21 +215,29 @@ def register_pytree_node(
             )
         )
     """
-    from ._pytree import _register_pytree_node
-
-    _register_pytree_node(
-        cls,
-        flatten_fn,
-        unflatten_fn,
-        serialized_type_name=serialized_type_name,
-    )
+    # TODO(XuehaiPan): remove this condition when we make Python pytree out-of-box support
+    # PyStructSequence types
+    if not optree.is_structseq_class(cls):
+        optree.register_pytree_node(
+            cls,
+            flatten_fn,
+            _reverse_args(unflatten_fn),
+            namespace=namespace,
+        )
 
-    optree.register_pytree_node(
-        cls,
-        flatten_fn,
-        _reverse_args(unflatten_fn),
-        namespace=namespace,
-    )
+    from . import _pytree as python
+
+    current_frame = inspect.currentframe()
+    previous_frame = current_frame.f_back if current_frame is not None else None
+    if previous_frame is not None and inspect.getmodule(previous_frame) is not python:
```

Comment:
This seems a bit dodgy. Can we get it to work without inspect? It seems like we should have a private `_private_register_pytree_node` APIs in both cxx_pytree and python_pytree and then have `_register_pytree_node` in both APIs call `cxx_pytree._private_register_pytree_node` and `python_pytree._private_register_pytree_node`. We can prevent people from using both of these directly by stating in the docstring what they should use.

Reply:
I have updated the PR.